grails-app/views/taglib/_globalSearch.gsp

```diff
@@ -1,41 +1,93 @@
+            <div class="dot-flashing"></div>
+        </div>
+        <span class="ri-close-line" id="global-search-close"></span>
+    </div>
+</g:form>
 <script>
-    $(document).ready(function() {
-        $("#${attrs.id}").autocomplete( {
-            delay: ${grailsApplication.config.openboxes.typeahead.delay},
-            minLength: ${grailsApplication.config.openboxes.typeahead.minLength},
-            source: function(req, resp) {
+  $(document)
+    .ready(function () {
+      $("#global-search-button")
+        .click(function () {
+          const globalSearchElement = $("#global-search").get(0);
+          const hiddenClassName = 'global-search--hidden';
+          if (globalSearchElement.classList.contains(hiddenClassName)) {
+            globalSearchElement.classList.remove(hiddenClassName)
+            $("#global-search-input").focus();
+          } else {
+            globalSearchElement.classList.add(hiddenClassName)
+          }
+        })
 
-				$.getJSON('${attrs.jsonUrl}', req, function(data) {
-					var suggestions = [];
-					$.each(data, function(i, item) {
-						suggestions.push(item);
-					});
-					resp(suggestions);
-				})
-            },
-            select: function(event, ui) {
-                window.location = ui.item.url;
-                return false;
-            },
-            focus: function(event, ui) {
-                this.value = ui.item.label;
-                event.preventDefault(); // Prevent the default focus behavior.
-            }
-        }).data("autocomplete")._renderItem = function( ul, item ) {
-          var link = $("<a></a>").text(item.label).css("color", item.color);
-          return $( "<li></li>" )
+      const splitMatchingStr = (data, str) => {
+        const indexOfMatched = data.indexOf(str);
+        if (indexOfMatched < 0) {
+          return { before: data };
+        }
+        const before = data.slice(0, indexOfMatched);
+        const matched = data.slice(indexOfMatched, indexOfMatched + str.length);
+        const after = data.slice(indexOfMatched + str.length, data.length);
+
+        return { before, matched, after };
+      };
+
+      function closeInputHandler () {
+        const hiddenClassName = 'global-search--hidden';
+        $("#global-search").get(0).classList.add(hiddenClassName);
+        $("#global-search-input").val('');
+      }
+
+      $("#global-search-close").click(closeInputHandler)
+
+      $("#global-search-input")
+        .blur(closeInputHandler)
+        .autocomplete({
+          delay: ${grailsApplication.config.openboxes.typeahead.delay},
+          minLength: ${grailsApplication.config.openboxes.typeahead.minLength},
+          position: { of: "#global-search" },
+          source: function (req, resp) {
+            $("#global-search-loading").get(0).classList.remove('d-none')
+            $.getJSON('${attrs.jsonUrl}', req, function (data) {
+              var suggestions = [];
```

Comment:
shouldn't we avoid using `var`? Wouldn't `const` work here?

Reply:
I did not notice it. 
This was legacy code so I did not pay much attention to this but in context of this functions I don't think it makes much difference using **var** ,**const** or **let**